addons/hr_holidays/models/hr_holidays.py

```diff
@@ -481,10 +493,10 @@ def action_validate(self):
                 holiday.write({'first_approver_id': current_employee.id})
             if holiday.holiday_type == 'employee' and holiday.type == 'remove':
                 holiday._validate_leave_request()
-            elif holiday.holiday_type == 'category':
+            elif holiday.holiday_type in ['category', 'department']:
                 leaves = self.env['hr.holidays']
-                for employee in holiday.category_id.employee_ids:
-                    values = holiday._prepare_create_by_category(employee)
+                for employee in holiday.category_id.employee_ids if holiday.holiday_type == 'category' else holiday.department_id.member_ids:
```

Comment:
I would 1/ compute employees 2/ perform the for. HAving the 2 in one line does not help understanding the code.

Reply:
hagrid